### DISCUSSION TITLE
Feature Ideas

### DISCUSSION THREAD

**Discussion by edfletcher** [nhexirc/nhex#1] [Ideas]
* Bouncer/ZNC support has already been mentioned by a few folks (thank you blueagain & InPhase)! Basic support should work OOTB as bouncers appear as normal IRC servers, but they do have an extended command set that is helpful for clients to implement.

**Comment by terrablue** [comment]
* Scripting languages (we mentioned bash-like/python/JS; I'd personally be also 100% content with just JS, but I think supporting different ones is attractive)
* If not already covered by scripts, nice ways to get around the icky parts of IRC (sending an invite without needing to op up, having that done for you automatically)
* Perhaps a nice interface for channel management, which can be often confusing (normal kicks VS akicks VS litharge -- the last one is Libera specific though)

**Comment by PantelicGR** [comment]
Greetings.
I would like this new irc client to support DCC tranfer file support like HexChat does.
Thanks in advance.

**Comment by PantelicGR** [comment]
- Support SASL connection to servers with the user having 2 options in the user interface:

1. manually entering the password of the user for each SASL connection without permanently saving the password and then autojoining the channels.
2. permanently saving the password of the user and not asking for it for each SASL connection and then autojoining the channels.

**Comment by edfletcher** [comment]
`dav1d` in ##rust made a request (thank you! ❤️) I'd already been thinking about and it's exactly why I added dev guidelines [2](https://github.com/nhexirc/client/blob/main/README.md?plain=1#L48) and [4](https://github.com/nhexirc/client/blob/main/README.md?plain=1#L50): ideally, in the long-term, **the front-end can be extracted into a standalone web app!**

If the backend is light and actor-modeled, swapping the "event bus" out for something different "should" be simple.

**Comment by edfletcher** [comment]
### UX theming

(Long-term)

It's all CSS anyway! Should be able to eventually allow full user customization.
